From 950796d68430c2a2778689ee00560d2f455937d4 Mon Sep 17 00:00:00 2001
From: David Brown <david.brown@linaro.org>
Date: Thu, 27 Jul 2017 10:01:23 -0600
Subject: [PATCH] good: rsa pkcs v1.5 good

This builds the bootloader and each image with the older PKCS v1.5
signature format.  Both the image and the upgrade should work.
---
 Makefile                | 2 +-
 samples/zephyr/Makefile | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1e9ec42..db97832 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ CFLAGS += -DMCUBOOT_SIGN_RSA -DMCUBOOT_USE_MBED_TLS
 
 # Newer uses should use the RSA-PSS signature algorithm.  This define
 # enables (and requires) this type of signature.
-#CFLAGS += -DMCUBOOT_RSA_PKCS1_15
+CFLAGS += -DMCUBOOT_RSA_PKCS1_15
 
 # ECDSA P-256
 #CONF_FILE = boot/zephyr/prj-p256.conf
diff --git a/samples/zephyr/Makefile b/samples/zephyr/Makefile
index 7678615..07bb71c 100644
--- a/samples/zephyr/Makefile
+++ b/samples/zephyr/Makefile
@@ -106,6 +106,7 @@ hello1: check
 		--header-size $(BOOT_HEADER_LEN) \
 		--align $(FLASH_ALIGNMENT) \
 		--version 1.2 \
+		--rsa-pkcs1-15 \
 		--included-header \
 		hello1/outdir/$(BOARD)/zephyr.bin \
 		signed-hello1.bin
@@ -125,6 +126,7 @@ hello2: check
 		--header-size $(BOOT_HEADER_LEN) \
 		--align $(FLASH_ALIGNMENT) \
 		--version 1.2 \
+		--rsa-pkcs1-15 \
 		--included-header \
 		--pad 0x60000 \
 		hello2/outdir/$(BOARD)/zephyr.bin \
-- 
2.11.0

